addrmgr/addrmanager.go

```diff
@@ -1070,13 +1044,17 @@ func (a *AddrManager) Good(addr *wire.NetAddress, subnetworkID *subnetworkid.Sub
 
 		// If this address was already tried, but subnetworkID was changed -
 		// update subnetworkID, than continue as though this is a new address
-		bucketList := a.addrTried[*oldSubnetworkID][triedBucketIndex]
-		for e := bucketList.Front(); e != nil; e = e.Next() {
-			if NetAddressKey(e.Value.(*KnownAddress).NetAddress()) == addrKey {
-				bucketList.Remove(e)
-				break
+		bucket := a.addrTried[*oldSubnetworkID][triedBucketIndex]
+		toRemoveIndex := -1
+		for i, ka := range bucket {
+			if NetAddressKey(ka.NetAddress()) == addrKey {
+				toRemoveIndex = i
 			}
 		}
+		if toRemoveIndex != -1 {
+			a.addrTried[*oldSubnetworkID][triedBucketIndex] =
```

Comment:
You can just move this to 1050 and break, and then get rid of weird -1

Reply:
I cannot, since that would be modifying the bucket while iterating over it.

Reply:
I would still prefer that you'll use a boolean